### PR TITLE
Add GetLastError API

### PR DESCRIPTION
**Pull request by FranklandJack**
Fixes https://github.com/oneapi-src/unified-runtime/issues/16.

### CODE REVIEW

**Comment by FranklandJack** [comment]
I think there is still some discussion to be had over the naming, but following discussions on #16 I've updated the entry point to match the semantics agreed upon there. @AidanBeltonS if you could take a look at this and confirm it matches your expectations that would be greatly appreciated.

File: include/ur_api.h [hunk 1]
```diff
@@ -1961,6 +1963,36 @@ urTearDown(
...
+/// 
+/// @returns
+///     - ::UR_RESULT_SUCCESS
+///     - ::UR_RESULT_ERROR_UNINITIALIZED
+///     - ::UR_RESULT_ERROR_DEVICE_LOST
+///     - ::UR_RESULT_ERROR_INVALID_NULL_POINTER
+///         + `NULL == pMessage`
+UR_APIEXPORT ur_result_t UR_APICALL
+urGetLastResult(
+    const char* pMessage                            ///< [out] optional pointer to a string containing adapter specific result
```
Comment (kbenzie) [inline include/ur_api.h:1992]: In PI this is `char**`, I don't think this will work. Also why is it optional?

File: include/ur_api.h [hunk 2]
```diff
@@ -1961,6 +1963,36 @@ urTearDown(
...
+///     - The string returned via the pMessage is a NULL terminated C style
+///       string.
+///     - The string returned via the pMessage is thread local.
+///     - The entry point will return UR_RESULT_SUCCESS if the result being
+///       reported is to be considered a warning. Any other result code returned
+///       indicates that the adapter specific result is an error.
+///     - The memory in the string returned via the pMessage is owned by the
+///       adapter.
+///     - The application may call this function from simultaneous threads.
+///     - The implementation of this function should be lock-free.
```
Comment (kbenzie) [inline include/ur_api.h:1982]: I think this should talk about using thread local storage explicitly in addition to these general statements about thread safety.

File: include/ur_api.h (hunk 1 above)
Comment (FranklandJack) [reply include/ur_api.h]: Added extra `*`. Removed optional.

**Review by AidanBeltonS — APPROVED** [review]
LGTM!

File: scripts/core/misc.yml [hunk 3]
```diff
@@ -22,3 +22,29 @@ params:
...
+       reported is to be considered a warning. Any other result code returned
+       indicates that the adapter specific result is an error."
+    -  "The memory in the string returned via the pMessage is owned by the
+       adapter."
+    - "The application may call this function from simultaneous
+       threads."
+    - "The implementation of this function should be lock-free."
+params:
+    - type: const char**
+      name: pMessage
```
Comment (kbenzie) [inline scripts/core/misc.yml:48]: I think this should be `ppMessage` by convention.

**Review by kbenzie — APPROVED** [review]

File: scripts/core/misc.yml (hunk 3 above)
Comment (FranklandJack) [reply scripts/core/misc.yml]: Ah! Good catch, forgot to update this, fixed now.

**Pull request merged**